### PR TITLE
Migrating repo to MCR

### LINKED ISSUE #223: Host images on mcr.microsoft.com
Steve Lasker wrote a blog back in May titled [Microsoft syndicates container catalog ](https://azure.microsoft.com/en-us/blog/microsoft-syndicates-container-catalog/), describing how Microsoft will start serving container images from mcr.microsoft.com, the Microsoft Container Registry (MCR), and syndicating the content describing the container images with Docker Hub and Red Hat.

When this transition is done, the images on Docker Hub will remain and serviced as to not cause any disruption to existing users.  All existing images will be copied to MCR and new tags will only be added to MCR.

The following is a working proposal of the repo hierarchy for the entire .NET family.

mcr.microsoft.com/dotnet (dotnet landing repo - no images)
mcr.microsoft.com/dotnet/framework (full FX landing repo – no images)
mcr.microsoft.com/dotnet/framework/runtime
mcr.microsoft.com/dotnet/framework/sdk
mcr.microsoft.com/dotnet/framework/aspnet
mcr.microsoft.com/dotnet/framework/wcf
mcr.microsoft.com/dotnet/framework/samples
mcr.microsoft.com/dotnet/core (core landing repo – no images)
mcr.microsoft.com/dotnet/core/runtime
mcr.microsoft.com/dotnet/core/runtime-deps
mcr.microsoft.com/dotnet/core/aspnet
mcr.microsoft.com/dotnet/core/sdk
mcr.microsoft.com/dotnet/core/samples
mcr.microsoft.com/dotnet/core-nightly (core nightly landing repo – no images)
mcr.microsoft.com/dotnet/core-nightly/runtime
mcr.microsoft.com/dotnet/core-nightly/runtime-deps
mcr.microsoft.com/dotnet/core-nightly/aspnet
mcr.microsoft.com/dotnet/core-nightly/sdk

Related to https://github.com/dotnet/dotnet-docker/issues/745

### PR DESCRIPTION
**Pull request by MichaelSimons**
Fixes #223

The engineering infra was mirrored from https://github.com/dotnet/dotnet-docker.

### CODE REVIEW

**Comment by MichaelSimons** [comment]
@dotnet-bot test this please

File: manifest.samples.json [hunk 1]
```diff
@@ -1,8 +1,9 @@
 {
+  "registry": "mcr.microsoft.com",
```
Comment (mthalman) [inline manifest.samples.json]: Should aspnetmvcapp be listed here as well?

File: samples/wcfapp/Dockerfile.web [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM microsoft/dotnet-framework:4.7.2-sdk AS build
+FROM mcr.microsoft.com/dotnet/framework/sdk:4.7.2 AS build
 WORKDIR /app
 COPY WcfServiceWebApp/. .
 RUN msbuild /p:Configuration=Release
```
Comment (mthalman) [inline samples/wcfapp/Dockerfile.web]: Add new line at bottom of file.

File: manifest.samples.json (hunk 1 above)
Comment (MichaelSimons) [reply manifest.samples.json]: IDK - possibly.  This is outside the scope of the MCR migration.  Will log a separate issue to track this.

File: samples/wcfapp/Dockerfile.web (hunk 2 above)
Comment (MichaelSimons) [reply samples/wcfapp/Dockerfile.web]: fixed

File: manifest.samples.json (hunk 1 above)
Comment (MichaelSimons) [reply manifest.samples.json]: https://github.com/Microsoft/dotnet-framework-docker/issues/238

**Pull request merged**